sql/core/src/main/scala/org/apache/spark/sql/execution/command/DataWritingCommand.scala

```diff
@@ -96,4 +98,23 @@ object DataWritingCommand {
       sparkContext.getLocalProperty(SQLExecution.EXECUTION_ID_KEY),
       metrics.values.toSeq)
   }
+
+  def assertEmptyRootPath(tablePath: URI, saveMode: SaveMode, sparkSession: SparkSession) {
+    if (saveMode != SaveMode.Overwrite &&
+      !sparkSession.sqlContext.conf.allowCreateTableAsSelectNonEmptyLocation) {
+      val filePath = new org.apache.hadoop.fs.Path(tablePath)
+      val fs = filePath.getFileSystem(sparkSession.sparkContext.hadoopConfiguration)
+      if(fs != null && fs.exists(filePath)) {
+        val locStats = fs.getFileStatus(filePath)
+        if(locStats != null && locStats.isDirectory) {
+          val lStats = fs.listStatus(filePath)
+          if(lStats != null && lStats.length != 0) {
```

Comment:
I think you can make it as a single condition:

```scala
if (fs.exists(filePath) &&
    fs.getFileStatus(filePath).isDirectory &&
    fs.listStatus(filePath).length != 0) {
  ...
}
```

Reply:
Done